src/ynEIGEN/RedemptionAssetsVault.sol

```diff
@@ -163,6 +164,19 @@ contract RedemptionAssetsVault is IRedemptionAssetsVault, Initializable, AccessC
      * @dev Iterates over the supported assets, transferring each asset's balance to the user until fully depleted.
      */
     function transferRedemptionAssets(address to, uint256 amount, bytes calldata /* data */) public onlyRedeemer whenNotPaused nonReentrant {
+        processRedemptionAssets(to, amount, processTransfer);
+    }
+
+    /** 
+     * @notice Withdraws a specified amount of redemption assets and processes them through ynETH.
+     * @param amount The amount of ETH to withdraw and process.
+     * @dev Requires the caller to be the redeemer and the contract to not be paused.
+     */
+    function withdrawRedemptionAssets(uint256 amount) public onlyRedeemer whenNotPaused nonReentrant {
+        processRedemptionAssets(address(ynEigen), amount, processWithdrawn);
+    }
+
+    function processRedemptionAssets(address to, uint256 amount, function(uint256, address, address) internal process) internal {
```

Comment:
add a generalized function called `processRedemptionAssets` that handles both

1. transferRedemptionAssets

This transfers assets to user on redemption and fees to fee collector

2. withdrawRedemptionAssets

this calls processWithdrawn on yneigen

they both do the same thing except on what they do to actually move assets out.